pinot-common/src/main/java/com/linkedin/pinot/common/partition/ReplicaGroupPartitionAssignmentGenerator.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Collections;
+import java.util.List;
+import org.apache.helix.AccessOption;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+
+/**
+ * Class to help to read, write, and generate replica group partition assignment
+ *
+ */
+public class ReplicaGroupPartitionAssignmentGenerator {
+
+  private ZkHelixPropertyStore<ZNRecord> _propertyStore;
+
+  public ReplicaGroupPartitionAssignmentGenerator(ZkHelixPropertyStore<ZNRecord> propertyStore) {
+    _propertyStore = propertyStore;
+  }
+
+  /**
+   * Read the replica group partition assignment from the property store
+   * @param tableNameWithType a table name
+   * @return Replica group partition assignment
+   */
+  public ReplicaGroupPartitionAssignment getReplicaGroupPartitionAssignment(String tableNameWithType) {
+    String path = ZKMetadataProvider.constructPropertyStorePathForInstancePartitions(tableNameWithType);
+    ZNRecord replicaGroupPartitionAssignment = _propertyStore.get(path, null, AccessOption.PERSISTENT);
+    ReplicaGroupPartitionAssignment partitionAssignment = null;
+    if (replicaGroupPartitionAssignment != null) {
+      partitionAssignment =
+          new ReplicaGroupPartitionAssignment(tableNameWithType, replicaGroupPartitionAssignment.getListFields());
+    }
+    return partitionAssignment;
+  }
+
+  /**
+   * Write the replica group partition assignment to property store
+   *
+   * @param partitionAssignment a replica group partition assignment
+   */
+  public void writeReplicaGroupPartitionAssignment(ReplicaGroupPartitionAssignment partitionAssignment) {
+    String tableNameWithType = partitionAssignment.getTableName();
+    ZNRecord znRecord = new ZNRecord(tableNameWithType);
+    znRecord.setListFields(partitionAssignment.getPartitionToInstances());
+    String path = ZKMetadataProvider.constructPropertyStorePathForInstancePartitions(tableNameWithType);
+    _propertyStore.set(path, znRecord, AccessOption.PERSISTENT);
+  }
+
+  /**
+   * Build the partition mapping table that maps a tuple of (partition number, replica group number) to a list of
+   * servers. Two important configurations are explained below.
+   *
+   * 'numInstancesPerPartition': this number decides the number of servers within a replica group.
+   * 'partitionColumn': this configuration decides whether to use the table or partition level replica groups.
+   *
+   * @param tableNameWithType: Name of table
+   * @param tableConfig: Configuration for table
+   * @return Replica group partition assignment from the given configuration
+   */
+  public ReplicaGroupPartitionAssignment buildReplicaGroupPartitionAssignment(String tableNameWithType,
+      TableConfig tableConfig, List<String> servers) {
+
+    // Fetch information required to build the mapping table from the table configuration.
+    ReplicaGroupStrategyConfig replicaGroupStrategyConfig = tableConfig.getValidationConfig().getReplicaGroupStrategyConfig();
```

Comment:
Can replicaGroupStrategyConfig be null? I see null checking only for segmentConfig in the place where buildReplicaGroupPartitionAssignment is called